### ISSUE TITLE
debug output gets mangled when it matches debug's formatter expression

### ISSUE THREAD

**Issue by capaj** [knex/knex#2783]
# Environment

Knex version: 0.15.2
Database + version: any
OS: any

as explained https://github.com/visionmedia/debug/issues/597#issue-353457717 when the string that is printed via `debug` package contains on of these: https://github.com/visionmedia/debug#formatters. Then it is passed to `util.format` which replaces it with `undefined`.

Note that in the demonstated case there was no transaction. If there was, it would get replaced with the transaction Id.

**Comment by elhigu** [comment]
Is this a bug or feature request or what?

**Comment by capaj** [comment]
bug

**Comment by elhigu** [comment]
Reproduction code is missing from report.

**Comment by capaj** [comment]
repro:
```js
const value = "jo"
db('users').where('username', 'ilike', `%${value}%`)
```

when I run that with `DEBUG=knex*`
I get this output:
```
knex:bindings [ 'undefinedo%' ]
```

expected output is:
```
knex:bindings [ '%jo%' ]
```

**Comment by elhigu** [comment]
Right... I suppose there is not much to do on knex side unless debug library adds support for disabling those formatters. Linking https://github.com/visionmedia/debug/issues/597 to be able to track its status easily.

**Comment by mercmobily** [comment]
Thank you for reporting this issue with the debug output getting mangled. This is indeed a problem with how the debug library handles certain string patterns that match its formatter expressions.

**The problem**
When Knex logs SQL bindings using the debug library, if the binding values contain patterns like `%j` (which is used for JSON formatting in debug), the debug library's `util.format` function replaces these patterns with `undefined`, causing the output to be corrupted.

**Example**
As you demonstrated, with a value like `"jo"` in a query:
```javascript
await db('users').where('username', 'ilike', `%${value}%`).select()
```

The debug output shows:
```
knex:bindings [ 'undefinedo%' ]
```

Instead of the expected:
```
knex:bindings [ '%jo%' ]
```

**Root cause**
This is a limitation of the debug library itself, which uses `util.format` that interprets `%j` as a JSON formatting specifier. The issue is tracked in debug-js/debug#597.

**Current status**
As elhigu noted in the original discussion, there's not much Knex can do about this directly since it's a limitation of the underlying debug library. The debug library would need to provide a way to disable these formatters or handle this case differently.

**Workarounds**
1. Use a different debug namespace that doesn't include the binding output
2. Implement custom logging instead of relying on the debug library for sensitive cases
3. Avoid using values that contain debug formatting patterns in your queries

**Future considerations**
If the debug library ever adds support for disabling formatters or provides a better solution for this edge case, Knex could potentially integrate that functionality. However, given that this issue has been open since 2018 without movement on the debug side, it's unlikely to be resolved soon.

For most practical purposes, this is a minor cosmetic issue that doesn't affect the actual query execution, only the debug output visibility.
